### PR TITLE
JACDAC

### PR DESCRIPTION
**Pull request by jamesadevine**
For visibility

### CODE REVIEW

File: source/types/Image.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@ DEALINGS IN THE SOFTWARE.
 #include "BitmapFont.h"
```
Comment (jamesadevine) [inline source/types/Image.cpp]: Blurgle flurgle, need to get rid of this bad change.

File: source/driver-models/Timer.cpp [hunk 2]
```diff
@@ -228,8 +228,12 @@ void Timer::sync(CODAL_TIMESTAMP t)
     // First, update our timestamps.
     currentTimeUs += t;
     overflow += t;
-    currentTime += overflow / 1000;
-    overflow = overflow % 1000;
+
```
Comment (jamesadevine) [inline source/driver-models/Timer.cpp]: @mmoskal Can you just double check this change (I think it works out).

File: source/driver-models/Display.cpp [hunk 3]
```diff
@@ -39,11 +39,12 @@ using namespace codal;
  *
  * @param id The id the display should use when sending events on the MessageBus. Defaults to DEVICE_ID_DISPLAY.
  */
-Display::Display(int width, int height, uint16_t id) : image(width, height)
+Display::Display(int width, int height, uint8_t ppb, uint16_t id) : image(width, height, ppb)
```
Comment (jamesadevine) [inline source/driver-models/Display.cpp]: To remove ppb.

File: source/driver-models/Timer.cpp (hunk 2 above)
Comment (mmoskal) [reply source/driver-models/Timer.cpp]: I think `overflow` used to be 64 bit, for which div/mod are always very slow. On M4 there's div instruction, and I guess mod must be implemented with it. On M0 there's is no div. Maybe add a comment that this is time critical, and does at most one loop iteration per millisecond? ;)

**Pull request merged**